java.js

```diff
@@ -68,53 +68,49 @@ export default class Java extends SdkmanTool {
         await this.validateVersion(expectedVersion)
 
         // If we got this far, we have successfully configured java.
-        core.setOutput(Java.tool, checkVersion)
+        this.outputInstalledToolVersion(Java.tool, checkVersion)
         this.info("java success!")
         return checkVersion
     }
 
-    // determines the desired version of java that is being requested. if the desired version
-    // presented to the action is present, that version is honored rather than the version
-    // presented in the .sdkmanrc file that can be optionally present in the checked out repo itself.
-    // Second value returned indicates whether or not the version returned has overridden
-    // the version from the .sdkmanrc file.
+    /**
+     * Determines the desired version of java that is being requested. if the desired version
+     * presented to the action is present, that version is honored rather than the version
+     * presented in the .sdkmanrc file that can be optionally present in the checked out repo itself.
+     * Second value returned indicates whether or not the version returned has overridden
+     * the version from the .sdkmanrc file.
+     * @param {string} actionDesiredVersion - This is the desired version of java as presented directly to the
+     * action.
+     * @returns {[string, boolean]} - The overall desired version of java that has been found, and the boolean
+     * indicates whether or not that version is overriding the value found in the .sdkmanrc file.
+     */
     getJavaVersion(actionDesiredVersion) {
```

Comment:
It seems that there are some duplication between `getJavaVersion` and `getKotlinVersion` could we just override the `getVersion` and put it on the SdkmanTool?

Reply:
Good point Alfonso, there is a good amount of duplication there. I will go about pushing a common method up to the superclass.